src/openms/source/QC/DBSuitability.cpp

```diff
@@ -290,9 +229,320 @@ namespace OpenMS
+    }
+
+    // temporary folder for search in- und output files
+    File::TempDir tmp_dir;
+    String mzml_path = tmp_dir.getPath() + "spectra.mzML";
+    String db_path = tmp_dir.getPath() + "database.FASTA";
+    String out_path = tmp_dir.getPath() + "out.idXML";
+
+    // override the in- and output files in the parameters
+    parameters.setValue(adapter_name + ":1:in", mzml_path);
+    parameters.setValue(adapter_name + ":1:database", db_path);
+    parameters.setValue(adapter_name + ":1:out", out_path);
+
+    // store data in temporary files
+    MzMLFile spectra_file;
+    spectra_file.store(mzml_path, exp);
+    FASTAFile database;
+    database.store(db_path, fasta_data);
+
+    String ini_path = tmp_dir.getPath() + "parameters.INI";
+    writeIniFile_(parameters, ini_path);
+
+    // run identification search
+    String proc_stdout;
+    String proc_stderr;
+    auto lam_out = [&](const String& out) { proc_stdout += out; };
+    auto lam_err = [&](const String& out) { proc_stderr += out; };
+
+    ExternalProcess ep(lam_out, lam_err);
+    OPENMS_LOG_DEBUG << "Running " << adapter_name << "..." << endl << endl;
+    const auto& rt = ep.run(adapter_name.toQString(), QStringList() << "-ini" << ini_path.toQString(), tmp_dir.getPath().toQString(), true);
+    if (rt != ExternalProcess::RETURNSTATE::SUCCESS)
+    { // error occured
+      OPENMS_LOG_ERROR << "An error occured while running " << adapter_name << "." << endl;
+      OPENMS_LOG_ERROR << "Standard output: " << proc_stdout << endl;
+      OPENMS_LOG_ERROR << "Standard error: " << proc_stderr << endl;
+      throw Exception::UnableToCreateFile(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Return state was: " + static_cast<Int>(rt));
+    }
+    // search was successful
+
+    // load result
+    vector<ProteinIdentification> prot_ids;
+    vector<PeptideIdentification> pep_ids;
+    IdXMLFile comet_out;
+    comet_out.load(out_path, prot_ids, pep_ids);
+
+    // annotate target/decoy information
+    PeptideIndexing indexer;
+    FASTAContainer<TFI_Vector> proteins(fasta_data);
+    OPENMS_LOG_DEBUG << "Running PeptideIndexer functionalities ..." << endl << endl;
+    OPENMS_LOG_INFO.remove(cout); // prevent indexer from writing statistic
+    PeptideIndexing::ExitCodes indexer_exit = indexer.run(proteins, prot_ids, pep_ids);
+    OPENMS_LOG_INFO.insert(cout); // revert logging change
+    if (indexer_exit != PeptideIndexing::ExitCodes::EXECUTION_OK)
+    {
+      OPENMS_LOG_ERROR << "An error occured while trying to index the search results." << endl;
+      throw Exception::UnableToCreateFile(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Return state was: " + static_cast<Int>(indexer_exit));
+    }
+
+    // calculate q-values
+    Param p;
+    p.setValue("use_all_hits", "true");
+    p.setValue("add_decoy_peptides", "true");
+    p.setValue("add_decoy_proteins", "true");
```

Comment:
we should really have a function `Param::overwriteValue` which checks if the value exists and fails otherwise....
This would make this much more safe...
@jpfeuffer , opinions?

Reply:
yes definitely. I often had problems when misspelling.

Reply:
this would also have the advantage that the description etc stays the same. Currently, setValue() removes it....